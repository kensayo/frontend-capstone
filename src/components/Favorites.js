import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchFavoritesfromUser } from '../redux/favorite/favorite';

const Favorites = () => {
  const dispatch = useDispatch();
  const user = useSelector((state) => state.userReducer.user);
  useEffect(() => {
    dispatch(fetchFavoritesfromUser(user.id));
  }, []);

  const favoritesState = useSelector((state) => state.favoritesReducer.favorites);

  return (
    <div className="pt-4 mt-4 pt-sd-2 mt-sd-2">
      <h1 className="pt-md-5 mt-md-5">My Favorites</h1>
      <ul className="list-group">
        {
        (favoritesState && user)
          ? favoritesState.map((fav) => (
            <li className="list-group-item" key={fav.id}>
              <p>
                {`
                  Computer brand  ${fav.brand}, 
                  hard disk drive ${fav.hdd} tera,
                  Intel Core ${fav.processor},
                   Ram: ${fav.ram},
                   Price: ${fav.price} dollars.
              `}

              </p>

            </li>

          ))
          : <div> Please signup or signin </div>
        }
      </ul>

    </div>
  );
};

export default Favorites;
