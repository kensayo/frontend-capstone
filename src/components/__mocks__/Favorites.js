import React from 'react';

const Favorites = (props) => {
  const {
    id, brand, hdd, processor, ram, price,
  } = props;
  return (
    <div className="pt-4 mt-4 pt-sd-2 mt-sd-2">
      <h1 className="pt-md-5 mt-md-5">My Favorites</h1>
      <ul className="list-group">
            <li className="list-group-item" key={id}>
              <p>
                {`
                  Computer brand  ${brand}, 
                  hard disk drive ${hdd} tera,
                  Intel Core ${processor},
                   Ram: ${ram},
                   Price: ${price} dollars.
              `}

              </p>

            </li>
            <div> Please signup or signin </div>

      </ul>

    </div>
  );
};

export default Favorites;