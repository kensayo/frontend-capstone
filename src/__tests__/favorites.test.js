import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import Favorites from '../components/Favorites';
import store from '../redux/configureStore';

test('matches home component snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Favorites />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
