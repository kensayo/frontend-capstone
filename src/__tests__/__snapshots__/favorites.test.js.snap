// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches home component snapshot 1`] = `
<div
  className="pt-4 mt-4 pt-sd-2 mt-sd-2"
>
  <h1
    className="pt-md-5 mt-md-5"
  >
    My Favorites
  </h1>
  <ul
    className="list-group"
  />
</div>
`;
